File: pkg/codegen/internal/test/testdata/azure-native-pp/azure-native.cs
using Pulumi;
using AzureNative = Pulumi.AzureNative;

class MyStack : Stack
{
    public MyStack()
    {
        var frontDoor = new AzureNative.Network.FrontDoor("frontDoor", new AzureNative.Network.FrontDoorArgs
        {
            RoutingRules = 
            {
                new AzureNative.Network.Inputs.RoutingRuleArgs
                {
                    RouteConfiguration = new AzureNative.Network.Inputs.ForwardingConfigurationArgs
                    {
                        OdataType = "#Microsoft.Azure.FrontDoor.Models.FrontdoorForwardingConfiguration",
                        BackendPool = new AzureNative.Network.Inputs.SubResourceArgs
                        {
                            Id = "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/frontDoors/frontDoor1/backendPools/backendPool1",
                        },
                    },
                },
            },
        });
        var endpoint = new AzureNative.Cdn.Endpoint("endpoint", new AzureNative.Cdn.EndpointArgs
        {
            DeliveryPolicy = new AzureNative.Cdn.Inputs.EndpointPropertiesUpdateParametersDeliveryPolicyArgs
            {
                Rules = 
                {
                    new AzureNative.Cdn.Inputs.DeliveryRuleArgs
                    {
                        Actions = 
                        {
                            new AzureNative.Cdn.Inputs.DeliveryRuleCacheExpirationActionArgs
                            {
                                Name = "CacheExpiration",
                                Parameters = new AzureNative.Cdn.Inputs.CacheExpirationActionParametersArgs
                                {
                                    CacheBehavior = "Override",
                                    CacheDuration = "10:10:09",
                                    CacheType = "All",
                                    OdataType = "#Microsoft.Azure.Cdn.Models.DeliveryRuleCacheExpirationActionParameters",
                                },
                            },
                            new AzureNative.Cdn.Inputs.DeliveryRuleResponseHeaderActionArgs
                            {
                                Name = "ModifyResponseHeader",
                                Parameters = new AzureNative.Cdn.Inputs.HeaderActionParametersArgs
                                {
                                    HeaderAction = "Overwrite",
                                    HeaderName = "Access-Control-Allow-Origin",
                                    OdataType = "#Microsoft.Azure.Cdn.Models.DeliveryRuleHeaderActionParameters",
                                    Value = "*",
                                },
                            },
                            new AzureNative.Cdn.Inputs.DeliveryRuleRequestHeaderActionArgs
                            {
                                Name = "ModifyRequestHeader",
                                Parameters = new AzureNative.Cdn.Inputs.HeaderActionParametersArgs
                                {
                                    HeaderAction = "Overwrite",
                                    HeaderName = "Accept-Encoding",
                                    OdataType = "#Microsoft.Azure.Cdn.Models.DeliveryRuleHeaderActionParameters",
                                    Value = "gzip",
                                },
                            },
                        },
                        Conditions = 
                        {
                            new AzureNative.Cdn.Inputs.DeliveryRuleRemoteAddressConditionArgs
                            {
                                Name = "RemoteAddress",
                                Parameters = new AzureNative.Cdn.Inputs.RemoteAddressMatchConditionParametersArgs
                                {
                                    MatchValues = 
                                    {
                                        "192.168.1.0/24",
                                        "10.0.0.0/24",
                                    },
                                    NegateCondition = true,
                                    OdataType = "#Microsoft.Azure.Cdn.Models.DeliveryRuleRemoteAddressConditionParameters",
                                    Operator = "IPMatch",
                                },
                            },
                        },
                        Name = "rule1",
                        Order = 1,
                    },
                },
            },
            EndpointName = "endpoint1",
            IsCompressionEnabled = true,
            IsHttpAllowed = true,
            IsHttpsAllowed = true,
            Location = "WestUs",
        });
    }

}

File: pkg/codegen/testing/test/testdata/azure-native-pp/go/azure-native.go
package main

import (
	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/cdn"
	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/network"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		_, err := network.NewFrontDoor(ctx, "frontDoor", &network.FrontDoorArgs{
			ResourceGroupName: pulumi.String("someGroupName"),
			RoutingRules: network.RoutingRuleArray{
				&network.RoutingRuleArgs{
					RouteConfiguration: network.ForwardingConfiguration{
						OdataType: "#Microsoft.Azure.FrontDoor.Models.FrontdoorForwardingConfiguration",
						BackendPool: network.SubResource{
							Id: "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/frontDoors/frontDoor1/backendPools/backendPool1",
						},
					},
				},
			},
		})
		if err != nil {
			return err
		}
		_, err = cdn.NewEndpoint(ctx, "endpoint", &cdn.EndpointArgs{
			Origins: cdn.DeepCreatedOriginArray{},
			DeliveryPolicy: &cdn.EndpointPropertiesUpdateParametersDeliveryPolicyArgs{
				Rules: []cdn.DeliveryRuleArgs{
					{
						Actions: pulumi.Array{
							{
								Name: "CacheExpiration",
								Parameters: {
									CacheBehavior: "Override",
									CacheDuration: "10:10:09",
									CacheType:     "All",
									OdataType:     "#Microsoft.Azure.Cdn.Models.DeliveryRuleCacheExpirationActionParameters",
								},
							},
							{
								Name: "ModifyResponseHeader",
								Parameters: {
									HeaderAction: "Overwrite",
									HeaderName:   "Access-Control-Allow-Origin",
									OdataType:    "#Microsoft.Azure.Cdn.Models.DeliveryRuleHeaderActionParameters",
									Value:        "*",
								},
							},
							{
								Name: "ModifyRequestHeader",
								Parameters: {
									HeaderAction: "Overwrite",
									HeaderName:   "Accept-Encoding",
									OdataType:    "#Microsoft.Azure.Cdn.Models.DeliveryRuleHeaderActionParameters",
									Value:        "gzip",
								},
							},
						},
						Conditions: pulumi.Array{
							{
								Name: "RemoteAddress",
								Parameters: {
									MatchValues: []string{
										"192.168.1.0/24",
										"10.0.0.0/24",
									},
									NegateCondition: true,
									OdataType:       "#Microsoft.Azure.Cdn.Models.DeliveryRuleRemoteAddressConditionParameters",
									Operator:        "IPMatch",
								},
							},
						},
						Name:  pulumi.String("rule1"),
						Order: pulumi.Int(1),
					},
				},
			},
			EndpointName:         pulumi.String("endpoint1"),
			IsCompressionEnabled: pulumi.Bool(true),
			IsHttpAllowed:        pulumi.Bool(true),
			IsHttpsAllowed:       pulumi.Bool(true),
			Location:             pulumi.String("WestUs"),
			ProfileName:          pulumi.String("profileName"),
			ResourceGroupName:    pulumi.String("resourceGroupName"),
		})
		if err != nil {
			return err
		}
		return nil
	})
}
